src/property.ts

```diff
@@ -953,7 +930,37 @@ function Property$ensureInited(property: Property, obj: Entity): void {
 
 		// Do not initialize calculated properties. Calculated properties should be initialized using a property get rule.
 		if (!property.isCalculated) {
-			Property$init(property, obj, Property$getInitialValue(property, obj));
+			let initData = null;
+			let initValue = null;
+
+			// Constant
+			if (property.isConstant)
+				initValue = typeof property.constant === "function" ? (property.constant = property.constant()) : property.constant;
+			else if (property.initializer) {
+				initValue = property.initializer.call(obj);
+				if (isEntityType(property.propertyType)) {
+					initData = initValue;
+					if (property.isList)
+						initValue = [];
+					else
+						initValue = new property.propertyType();
```

Comment:
When `init` is used for a reference property, establish the property value (array or new object) prior to using the actual data, since it could cause rules to run which could impact other objects.